api/app/Http/Controllers/BaseController.php

```diff
@@ -34,61 +34,171 @@
     protected $transformer = 'App\Http\Transformers\BaseTransformer';
 
     /**
-     * Transform a collection for response.
+     * Get all entities.
      *
-     * @param  Collection $collection
-     * @return array
+     * @return Response
      */
-    protected function collection(Collection $collection)
+    public function getAll()
     {
-        $transformer = App::make('App\Services\Transformer');
+        return $this->collection($this->repository->all());
+    }
 
-        return $transformer->collection($collection, new $this->transformer);
+    /**
+     * Get one entity.
+     *
+     * @param  string  $id
+     * @return Response
+     */
+    public function getOne($id)
+    {
+        return $this->item($this->repository->find($id));
     }
 
     /**
-     * Transform an item for response.
+     * Post a new entity.
      *
-     * @param  Model  $item
-     * @return array
+     * @param  Request $request
+     * @return mixed
      */
-    protected function item(Model $item)
+    public function postOne(Request $request)
     {
-        $transformer = App::make('App\Services\Transformer');
+        if (!$this->validator->with($request->all())->passes()) {
+            return $this->validator->errors();
+        }
 
-        return $transformer->item($item, new $this->transformer);
+        return response($this->repository->create($request->all()), 201);
     }
 
     /**
-     * Get all entities.
+     * Put an entity.
      *
+     * @param  string   $id
+     * @param  Request  $request
      * @return Response
      */
-    public function getAll()
+    public function putOne($id, Request $request)
     {
-        return $this->collection($this->repository->all());
+        if (!$this->validator->put()->with(array_add($request->all(), 'entity_id', $id))->passes()) {
```

Comment:
This validation check should verify that the `$request->all()[Model->primaryKey] === $id` and if not throw a validation exception. 

The integration test currently fails if the *json id is valid* and the *route id is invalid*. Basically if they don't match we should reject it as the client has clearly made a mistake somewhere.